server/utils/TestingUtils/TestDatabaseHandler.js

```diff
@@ -0,0 +1,48 @@
+const mongoose = require('mongoose');
+const { MongoMemoryServer } = require('mongodb-memory-server');
+
+const mongooseOpts = {
+  useNewUrlParser: true,
+  useUnifiedTopology: true,
+};
+
+const mongod = new MongoMemoryServer({
+  instance: {
+    dbName: 'ark-automate',
+    port: 59051,
+  },
+});
+
+/**
+ * Connect to the in-memory database.
+ */
+module.exports.connect = async () => {
+  const uri = await mongod.getUri();
+  await mongoose.createConnection(uri, mongooseOpts);
+  await mongoose.connect(uri, mongooseOpts);
+};
+
+/**
+ * Drop database, close the connection and stop mongod.
+ */
+module.exports.closeDatabase = async () => {
+  await mongoose.connection.dropDatabase();
+  await mongoose.connection.close();
+  await mongoose.disconnect();
+  await mongod.stop();
+};
+
+/**
+ * Remove all the data for all db collections.
+ */
+module.exports.clearDatabase = async () => {
+  const { collections } = mongoose.connection;
+
+  // fix according to https://docs.w3cub.com/eslint/rules/no-await-in-loop.html
```

Comment:
does this need to be fixed or is this the fix?

Reply:
This is the fix for the es-lint no-await-in-loop error, by simply postponing the fulfillment of the promises to the end